### PR TITLE
Install Hashicorp repo packages when available

### PR DESCRIPTION
**Pull request by jorhett**
@bbaassssiiee this allows use of the Hashicorp-provided package repos

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -16,8 +16,17 @@
...
-  include: boundary_install.yml
+- name: installing Boundary on Debian
+  include: install_package_debian.yml
+  when: ansible_os_family == 'Debian' and boundary_install_package == true
+
+- name: installing Boundary on RedHat
+  include: install_package_redhat.yml
+  when: ansible_os_family == 'RedHat' and boundary_install_package == true
+
+- name: installing from source
```
Comment (bbaassssiiee) [inline tasks/main.yml]: Could this be renamed to `downloading binary`? There is no source code involved. (Please the auditors)

File: defaults/main.yml [hunk 2]
```diff
@@ -1,18 +1,23 @@
 ---
-# yamllint disable rule:line-length
-boundary_user: 'boundary'
-boundary_group: 'boundary'
-boundary_create_account: true
+# Package installation
+boundary_install_package: true
+boundary_version: '0.1.0'
```
Comment (bbaassssiiee) [inline defaults/main.yml:4]: Please increment to 0.1.2

File: defaults/main.yml [hunk 3]
```diff
@@ -1,18 +1,23 @@
 ---
-# yamllint disable rule:line-length
```
Comment (bbaassssiiee) [inline defaults/main.yml:2]: Please change this to `# yamllint disable-line rule:line-length` so it applies to the next line and not the whole file.

File: defaults/main.yml [hunk 4]
```diff
@@ -1,18 +1,23 @@
...
 boundary_controller_file: '{{ boundary_home_directory }}/boundary-controller.hcl'
-boundary_autocomplete_state: file # can be changed to absent
+boundary_install_directory: "'/usr/bin' if boundary_installl_package == true else '{{ boundary_data_directory }}/bin'"
 
+# Only used if package installation == false
 boundary_url: 'https://releases.hashicorp.com/boundary'
-boundary_version: '0.1.0'
+boundary_data_directory: '/opt/boundary'
+
+boundary_user: 'boundary'
```
Comment (bbaassssiiee) [inline defaults/main.yml:17]: These lines only moved, inserting new code should not change existing lines.

File: tasks/install_package_redhat.yml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+---
+- name: add Hashicorp repo
+  command: yum-config-manager --add-repo=yum-config-manager --add-repo https://rpm.releases.hashicorp.com/rhel/hashicorp.repo
+  args:
+    creates: /etc/yum.repos.d/hashicorp.repo
+  become: true
+
+- name: install Boundary
+  rpm:
```
Comment (bbaassssiiee) [inline tasks/install_package_redhat.yml]: Please use `package` or `yum` here, **rpm** is not an ansible module.

**Review by bbaassssiiee — CHANGES_REQUESTED** [review]
Thanks for this addition, it is a good idea to install the package when available, but still have the user decide.

I request a couple of small changes. Please rebase your fork and look at the tests.

File: defaults/main.yml (hunk 4 above)
Comment (jorhett) [reply defaults/main.yml:17]: I was attempting to bring up variables people might want to change, out of a cloud of variables that will only be used if installing manually ... completely open to other approaches, but desire to make the things a person would be most likely to change right up front.

File: tasks/install_package_redhat.yml (hunk 5 above)
Comment (jorhett) [reply tasks/install_package_redhat.yml]: fixed

File: tasks/main.yml (hunk 1 above)
Comment (jorhett) [reply tasks/main.yml]: done

File: defaults/main.yml (hunk 2 above)
Comment (jorhett) [reply defaults/main.yml:4]: done

File: defaults/main.yml (hunk 3 above)
Comment (jorhett) [reply defaults/main.yml:2]: done

File: defaults/main.yml (hunk 4 above)
Comment (bbaassssiiee) [reply defaults/main.yml:17]: That makes sense.

**Review by bbaassssiiee — APPROVED** [review]
Thanks!

**Pull request merged**